### PR TITLE
Ball in Maze game added

### LINKED ISSUE #4205: [New game]: Ball-in-Maze Game
### 🎮 Game Request

Game logic and basic description:Ball in maze is a game to manipulating the maze board to get the ball towards the goal.

### Point down the features

Game points:
In Ball-in-Maze, users can:

* Start, pause, restart
* Move up, down, left, right
* Restart the game if ball dropped
* Modal for pause, rules.

### Select program in which you are contributing

GSSoC24

### Code of Conduct

- [X] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) of this project.

### PR DESCRIPTION
**Pull request by Nayanika1402**
## PR Description 📜

Please include summary related to the issue you have fixed and describe your PR in brief over here by specifying the issue number on which you were working below
Fixes #4205 

<hr>
 
## Mark the task you have completed ✅

- [x] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) & [CODE OF CONDUCT](https://github.com/kunjgit/GameZone/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generates no new warnings.
- [x] I have followed proper naming convention showed in [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md)
- [x] I have added screenshot for website preview in assets/images 
- [x] I have added entries for my game in main README.md
- [x] I have added README.md in my folder 
- [x] I have added working video of the game in README.md (optional)
- [x] I have specified the respective issue number for which I have requested the new game.

<hr>

## Add your screenshots(Optional) 📸

Ball_in_Maze

--- 
<br>

## Thank you soo much for contributing to our repository 💗

### CODE REVIEW

**Comment by kunjgit** [comment]
Thank you @Nayanika1402 ,for creating the PR and contributing to our GameZone 💗 
 Review team will review the PR and will reach out to you soon! 😇 
 Make sure that you have marked all the tasks that you are done with ✅. 
 Thank you for your patience! 😀

**Comment by Nayanika1402** [comment]
@kunjgit Kindly review my pr.

**Comment by Nayanika1402** [comment]
@kunjgit Please merge my pr.You have not reviewed my pr since long on the other hand you are accepting other's prs.

**Comment by ishita-43** [comment]
Conflicts are arising resolve them!!

**Comment by Nayanika1402** [comment]
@ishita-43 Please now kindly merge it,I have resolved the conflicts.

File: README.md [hunk 1]
```diff
@@ -330,6 +330,8 @@ This repository also provides one such platforms where contributers come over an
 |[Chess_Game_computer](https://github.com/kunjgit/GameZone/tree/main/Games/Chess_Game_computer)   |
 |[Turn_on_the_light](https://github.com/kunjgit/GameZone/tree/main/Games/Turn_on_the_light)   |
 | [Tic-Tac-Toe Game](https://github.com/kunjgit/GameZone/tree/main/Games/Tic-Tac-Toe) |
+| [Rapid_click_frenzy](https://github.com/kunjgit/GameZone/tree/main/Games/Rapid_click_frenzy)
```
Comment (ishita-43) [inline README.md:333]: Only add entry of your game and remove this one rapid_click frenzy one!!

File: assets/js/gamesData.json [hunk 2]
```diff
@@ -2102,6 +2102,13 @@
  "gameTitle": "Guess_The_Song",
   "gameUrl": "Guess_The_Song",
   "thumbnailUrl": "Guess_The_Song.png"
+},
+"417":{
+  "gameTitle": "Ball in Maze",
+   "gameUrl": "Ball_in_Maze",
+   "thumbnailUrl": "Ball_in_Maze.png"
+}
+=======
```
Comment (ishita-43) [inline assets/js/gamesData.json:2111]: remove this line if not necessary!!

**Review by ishita-43 — CHANGES_REQUESTED** [review]
Please do requested changes!!

**Comment by Nayanika1402** [comment]
@ishita-43 Requested changes done.

File: assets/js/gamesData.json [hunk 3]
```diff
@@ -2102,6 +2102,12 @@
  "gameTitle": "Guess_The_Song",
   "gameUrl": "Guess_The_Song",
   "thumbnailUrl": "Guess_The_Song.png"
+},
+"417":{
```
Comment (ishita-43) [inline assets/js/gamesData.json:2106]: this should not be numbered same as the below one please refer other entries!!

**Review by ishita-43 — CHANGES_REQUESTED** [review]
change this one also!!

File: assets/js/gamesData.json (hunk 3 above)
Comment (Nayanika1402) [reply assets/js/gamesData.json]: @ishita-43 Done as requested.

**Comment by Nayanika1402** [comment]
@ishita-43 Please merge my pr.

File: assets/js/gamesData.json [hunk 4]
```diff
@@ -2108,10 +2108,24 @@
    "gameUrl": "Brick Buster",
    "thumbnailUrl": "Brick.png"
  },
-  "418":{
+  "421":{
```
Comment (ishita-43) [inline assets/js/gamesData.json:2111]: this should be 17 instead and follow pattern respectively for below

File: assets/js/gamesData.json [hunk 5]
```diff
@@ -2108,10 +2108,24 @@
...
   "gameTitle": "Ball in Maze",
    "gameUrl": "Ball_in_Maze",
    "thumbnailUrl": "Ball_in_Maze.png"
+  },
+ "417":{
```
Comment (ishita-43) [inline assets/js/gamesData.json:2116]: update this one to 418

File: assets/js/gamesData.json [hunk 6]
```diff
@@ -2108,10 +2108,24 @@
...
+  "gameTitle": "Soccer",
+  "gameUrl": "Soccer",
+  "thumbnailUrl": "Soccer"
+},
+"409":{"gameTitle": "Pen_Pointer_Fight",
+"gameUrl": "PenPointerFight",
+"thumbnailUrl": "PenPointerFight.png"
+},
+"418":{
```
Comment (ishita-43) [inline assets/js/gamesData.json:2125]: rename as 419

**Review by ishita-43 — CHANGES_REQUESTED** [review]

File: assets/js/gamesData.json (hunk 6 above)
Comment (Nayanika1402) [reply assets/js/gamesData.json]: @ishita-43 Resolved.

File: assets/js/gamesData.json [hunk 7]
```diff
@@ -2108,12 +2108,12 @@
...
    "thumbnailUrl": "Brick.png"
  },
-  "421":{
+  "417":{
   "gameTitle": "Ball in Maze",
    "gameUrl": "Ball_in_Maze",
    "thumbnailUrl": "Ball_in_Maze.png"
   },
- "417":{
+ "418":{
```
Comment (ishita-43) [inline assets/js/gamesData.json:2116]: label as 409

File: assets/js/gamesData.json [hunk 8]
```diff
@@ -2122,7 +2122,7 @@
 "gameUrl": "PenPointerFight",
 "thumbnailUrl": "PenPointerFight.png"
 },
-"418":{
+"419":{
```
Comment (ishita-43) [inline assets/js/gamesData.json:2125]: as 418

**Review by ishita-43 — CHANGES_REQUESTED** [review]

File: assets/js/gamesData.json (hunk 7 above)
Comment (Nayanika1402) [reply assets/js/gamesData.json]: Done @ishita-43

File: README.md [hunk 9]
```diff
@@ -404,6 +404,6 @@ Terms and conditions for use, reproduction and distribution are under the [Apach
 </a>
 </center>
 <br>
+<p align="right"><a href="#top">Back to top</a></p>
```
Comment (ishita-43) [inline README.md:407]: why you duplicates this if not needed remove

**Review by ishita-43 — CHANGES_REQUESTED** [review]

File: README.md (hunk 9 above)
Comment (ishita-43) [reply README.md:407]: we are not looking for more and more pr more than the quality of the pr. For us quality of pr is as important as pr merging for you.

**Review by ishita-43 — APPROVED** [review]
Good work!!

**Comment by ishita-43** [comment]
PR will be merged by Admin soon till then thank you for your patience.

**Comment by kunjgit** [comment]
@Nayanika1402 resolve merge conflicts !

**Comment by kunjgit** [comment]
Thank you  @Nayanika1402 , for your valuable time and contribution in our GameZone 💗. 
 It’s our GameZone, so Let’s build this GameZone altogether !!🤝
  Hoping to see you soon with another PR again 😇 
 Wishing you all the best for your journey into Open Source🚀

**Pull request merged**